WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java

```diff
@@ -499,25 +499,23 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         switch (requestCode) {
             case RequestCodes.PICTURE_LIBRARY:
             case RequestCodes.VIDEO_LIBRARY:
-            case RequestCodes.FILE_LIBRARY:
             case RequestCodes.AUDIO_LIBRARY:
+                handlePickerResult(data, resultCode);
+                break;
+            case RequestCodes.FILE_LIBRARY:
                 if (resultCode == Activity.RESULT_OK && data != null) {
-                    if (data.hasExtra(MediaPickerConstants.EXTRA_MEDIA_URIS)) {
-                        List<Uri> uris = convertStringArrayIntoUrisList(
-                                data.getStringArrayExtra(MediaPickerConstants.EXTRA_MEDIA_URIS));
-                        for (Uri uri : uris) {
-                            getMediaFromDeviceAndTrack(uri, requestCode);
-                        }
+                    if (WPMediaUtils.shouldAdvertiseImageOptimization(this)) {
```

Comment:
🚫 This check is causing a bug where you can only upload the image when the pop-up appears since it doesn't have an else clause simply calling `handlePickerResult`.

```kotlin
if (WPMediaUtils.shouldAdvertiseImageOptimization(this)) {
    WPMediaUtils.advertiseImageOptimization(this, () -> handlePickerResult(data, resultCode));
} else {
    handlePickerResult(data, resultCode);
}
```

Reply:
Ouch, thank you so much for spotting Thomas, can't believe I let that slip through. Addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/19556/commits/29404be05325314ccbeff5623cc423d0106eb6b9. 🤦‍♀️